### PR TITLE
Finally add DocStringExtensions 🥳 

### PR DESCRIPTION
**Pull request by stelmo**
This should reduce docstring out-of-sync issues!

### CODE REVIEW

**Comment by laurentheirendt** [comment]
this is cool 👍🏼

File: src/base/logging/log.jl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 """
-    macro _make_logging_group(sym::Symbol, doc::String)
+$(TYPEDSIGNATURES)
```
Comment (exaexa) [inline src/base/logging/log.jl:3]: Q: is there any way to use TYPEDSIGNATURES in the macro that actually generates the functions (below in this file)?

File: src/base/types/abstract/Maybe.jl [hunk 2]
```diff
@@ -7,7 +7,7 @@ A nice name for "nullable" type.
 const Maybe{T} = Union{Nothing,T}
```
Comment (exaexa) [inline src/base/types/abstract/Maybe.jl:7]: Q: why no DSE here?

File: src/base/macros/model_wrapper.jl [hunk 3]
```diff
@@ -1,6 +1,6 @@
 
 """
-    _inherit_model_methods_impl(mtype::Symbol, arglist, access, fwdlist, fns...)
+$(TYPEDSIGNATURES)
 
 A helper backend for [`@_inherit_model_methods`](@ref) and
 [`@_inherit_model_methods_fn`](@ref).
```
Comment (exaexa) [inline src/base/macros/model_wrapper.jl:6]: Q: the macros lower in this file can't be documented with DSE?

(Also, would it be possible to use DSE again for the generated docstring below?)

**Review by exaexa — COMMENTED** [review]
A few questions but otherwise looks okay

extra questions:
- what about the extra operators in `src/reconstruction/Reactions.jl`? (could we have the typed signature and print out the intended inline usage in an example?)
- what about code generated by macros, is it possible to smuggle TYPEDSIGNATURES in there?

File: src/base/types/abstract/Maybe.jl (hunk 2 above)
Comment (stelmo) [reply src/base/types/abstract/Maybe.jl:7]: I did not DSE anything constants, not sure they have something for it?

**Comment by stelmo** [comment]
> Ok this is a good job. I've got a few questions but otherwise looks okay.
> 
> * what about the extra operators in `src/reconstruction/Reactions.jl`? (could we have the typed signature and print out the intended inline usage in an example?)

They looked weird so I left them, but tbh, there is no good reason for leaving them. Adding them! 

> * what about code generated by macros (e.g. in `src/base/macros/model_wrapper.jl`), is it possible to smuggle `TYPEDSIGNATURES` in there?

Inserting docstrings inside a macro that generates documentation seems to be an issue..

> * is there no way to print out the `const` typedefs (e.g. in `src/base/types/abstract/MetabolicModel.jl`)

I did not find anything for constants... Seems they will just be manual

File: src/base/types/abstract/Maybe.jl (hunk 2 above)
Comment (exaexa) [reply src/base/types/abstract/Maybe.jl:7]: ah this is the case of "constants docs", nvm.

**Review by exaexa — APPROVED** [review]
All answered I guess. Re the operators, since they are user-facing and the involved types are completely internal I guess we can do without the precise doc.

**Pull request merged**